### PR TITLE
feat(YTPLR): add Presence

### PR DESCRIPTION
**Pull request by PeterStrick**
## Description 
Adds a Presence for YTPLR (YouTube Playlist Randomizer)

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

YTPLR Modification
</details>

<details>
<summary> Proof showing the Website and Discord Profile without the Presence </summary>

YTPLR WithoutPresence
</details>

<details>
<summary> Proof showing the Website and Discord Profile with the Presence enabled </summary>

YTPLR WithPresence BetaExtension
</details>

<details>
<summary> Screenshot of the Settings Window </summary>

YTPLR Settings
</details>

### CODE REVIEW

File: websites/Y/YTPLR/metadata.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+		"name": "uwu_peter",
+		"id": "762219738570555412"
+	},
+	"service": "YTPLR",
+	"description": {
+		"en": "Shuffles a YouTube Playlist in random order"
+	},
+	"url": "youtube-playlist-randomizer.bitbucket.io",
+	"version": "1.0.0",
+	"logo": "https://youtube-playlist-randomizer.bitbucket.io/music-icon.png",
```
Comment (theusaf) [inline websites/Y/YTPLR/metadata.json:14]: Logo must be 512x512px. You may use tools like waifu2x to upscale if needed.

File: websites/Y/YTPLR/presence.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
+const presence = new Presence({
+		clientId: "1284161421957136486",
+	}),
+	strings = presence.getStrings({
+		play: "Watching",
+		pause: "Paused",
+	}),
```
Comment (theusaf) [inline websites/Y/YTPLR/presence.ts:7]: See the [localization repository](https://github.com/PreMiD/Localization/blob/main/src/Presence/general.json) for keys to use for strings.

File: websites/Y/YTPLR/presence.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	Logo = "https://youtube-playlist-randomizer.bitbucket.io/music-icon.png",
+}
+
+presence.on("iFrameData", (data: DataInterface) => {
+	videoTitle = data.title;
+	videoCurrentTime = data.currentTime;
+	videoDuration = data.duration;
+	videoPaused = data.paused;
+	videoPlaylistTotal = parseInt(document.querySelector("#all").textContent);
+	videoPlaylistID = (document.querySelector("#pid") as HTMLInputElement).value;
```
Comment (theusaf) [inline websites/Y/YTPLR/presence.ts:35]:
```suggestion
	videoPlaylistID = document.querySelector<HTMLInputElement>("#pid").value;
```

File: websites/Y/YTPLR/presence.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+		},
+		[startTimestamp, endTimestamp] = presence.getTimestamps(
+			Math.floor(videoCurrentTime),
+			Math.floor(videoDuration)
+		),
+		[showButtons] = await Promise.all([
+			presence.getSetting<boolean>("buttons"),
+		]);
```
Comment (theusaf) [inline websites/Y/YTPLR/presence.ts:49]: As there is only one setting, `Promise.all` is not needed here.
```suggestion
		showButtons = await presence.getSetting<boolean>("buttons");
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/Y/YTPLR/metadata.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+			"id": "buttons",
+			"title": "Show Buttons",
+			"icon": "fas fa-compress-arrows-alt",
+			"value": true
+		}
+	],
+	"category": "videos",
+	"iFrameRegExp": ".*",
+	"iframe": true,
+	"readLogs": false
```
Comment (theusaf) [inline websites/Y/YTPLR/metadata.json:32]: This field is by default `false`
```suggestion
```

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

File: websites/Y/YTPLR/metadata.json [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+	"settings": [
+		{
+			"id": "buttons",
+			"title": "Show Buttons",
+			"icon": "fas fa-compress-arrows-alt",
+			"value": true
+		}
+	],
+	"category": "videos",
+	"iFrameRegExp": ".*",
```
Comment (Bas950) [inline websites/Y/YTPLR/metadata.json:30]: Are you able to narrow this regex?

File: websites/Y/YTPLR/presence.ts [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+interface DataInterface {
+	title: string;
+	currentTime: number;
+	duration: number;
+	paused: boolean;
+}
+
+const enum Assets {
+	Logo = "https://youtube-playlist-randomizer.bitbucket.io/music-icon.png",
```
Comment (Bas950) [inline websites/Y/YTPLR/presence.ts:26]:
```suggestion
	Logo = "https://i.imgur.com/hsDOqgw.png",
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by Bas950** [comment]
Also maybe add an altname for the full name

File: websites/Y/YTPLR/metadata.json (hunk 6 above)
Comment (PeterStrick) [reply websites/Y/YTPLR/metadata.json:30]: I can try to do so tomorrow

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**